src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -668,9 +684,23 @@ static bool pick_from_internal_rr_locked(
 
 static grpc_lb_policy_args *lb_policy_args_create(grpc_exec_ctx *exec_ctx,
                                                   glb_lb_policy *glb_policy) {
-  grpc_lb_addresses *addresses =
-      process_serverlist_locked(exec_ctx, glb_policy->serverlist);
+  GPR_ASSERT(glb_policy->serverlist != NULL || 
+             glb_policy->fallback_backend_addresses != NULL);
+  grpc_lb_addresses *addresses;
+  if (glb_policy->serverlist != NULL) {
+    GPR_ASSERT(glb_policy->serverlist->num_servers > 0);
+    addresses = process_serverlist_locked(exec_ctx, glb_policy->serverlist);
+  } else if (glb_policy->fallback_backend_addresses != NULL) {
```

Comment:
Can probably remove the `if` here -- if we get here and do not have a response from the balancer, then we should unconditionally use the addresses returned by the resolver.